core/shared/platform/common/posix/posix_socket.c

```diff
@@ -884,7 +884,7 @@ os_socket_set_ip_ttl(bh_socket_t socket, uint8_t ttl_s)
 int
 os_socket_get_ip_ttl(bh_socket_t socket, uint8_t *ttl_s)
 {
-    socklen_t opt_len = sizeof(ttl_s);
+    socklen_t opt_len = sizeof(*ttl_s);
```

Comment:
According to `os_socket_set_ip_ttl` above, L877, I think it should be `sizeof(uint8_t)`, but I am not very sure.
@loganek could you help check whether the size here is `sizeof(uint8_t)`? The issue is reported by CodeQL introduced by PR #2812:
https://github.com/bytecodealliance/wasm-micro-runtime/security/code-scanning/70
https://github.com/bytecodealliance/wasm-micro-runtime/security/code-scanning/69

Reply:
sockopt is an ancient api with a lot of confusions and compatibility hacks.

`man 4 ip` on macOS has an example like:
```
     int ttl = 60;                   /* max = 255 */
     setsockopt(s, IPPROTO_IP, IP_TTL, &ttl, sizeof(ttl));
```
```
     u_char ttl;     /* range: 0 to 255, default = 1 */
     setsockopt(s, IPPROTO_IP, IP_MULTICAST_TTL, &ttl, sizeof(ttl));
```

otoh,
https://www.man7.org/linux/man-pages/man7/ip.7.html says:
(see the last sentence)
```
       IP_MULTICAST_TTL (since Linux 1.2)
              Set or read the time-to-live value of outgoing multicast
              packets for this socket.  It is very important for
              multicast packets to set the smallest TTL possible.  The
              default is 1 which means that multicast packets don't
              leave the local network unless the user program explicitly
              requests it.  Argument is an integer.
```

https://www.ibm.com/docs/en/i/7.1?topic=ssw_ibm_i_71/apis/ssocko.htm even says:
```
IP_TTL
....
Note:Only the rightmost octet (least significant octet) of the integer value is used.
```

Reply:
I think we should be passing `sizeof(*ttl_s)` (regardless of what type is that. I'm a bit worried that under the hood, some of the implementations will try to read as much as we pass as a size - if we pass `sizeof(int)`, three bytes will contain a garbage. If we really want this to be four bytes long, we should rather change the type of the parameter to be `uint32_t*`, but looks like most of the implementations expect the value to be within 0-255 range anyway.

Reply:
Yes, it's a little confusing, what the size should be may depend on which option to set/get and what the platform is. But IIUC, for the same option and same platform, the size to set/get should be the same, if we change the size in os_socket_get_ip_ttl to `sizeof(int)`, then we should also change the size in os_socket_set_ip_ttl to `sizeof(int)` accordingly.

Reply:
> I think we should be passing `sizeof(*ttl_s)` (regardless of what type is that. I'm a bit worried that under the hood, some of the implementations will try to read as much as we pass as a size - if we pass `sizeof(int)`, three bytes will contain a garbage. If we really want this to be four bytes long, we should rather change the type of the parameter to be `uint32_t*`, but looks like most of the implementations expect the value to be within 0-255 range anyway.

Yes, agree.